src/sentry/migrations/0728_backfill_groupsearchviews_with_pinned_searches.py

```diff
@@ -0,0 +1,68 @@
+# Generated by Django 5.0.6 on 2024-06-03 17:27
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.models.savedsearch import SortOptions
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_groupsearchviews_with_pinned_searches(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    PRIORITIZED_QUERY = "is:unresolved issue.priority:[high, medium]"
+    GroupSearchView = apps.get_model("sentry", "GroupSearchView")
+    SavedSearch = apps.get_model("sentry", "SavedSearch")
+
+    pinned_searches = SavedSearch.objects.filter(visibility="owner_pinned")
+    for pinned_search in RangeQuerySetWrapperWithProgressBar(pinned_searches):
+        GroupSearchView.objects.update_or_create(
+            organization=pinned_search.organization,
+            user_id=pinned_search.owner_id,
+            position=0,
+            defaults={
+                "name": "Default Search",
+                "query": pinned_search.query,
+                "query_sort": pinned_search.sort,
+            },
+        )
+        GroupSearchView.objects.create(
```

Comment:
Should this also be `update_or_create`? Just in case this needs to be re-run, then at least it'd idempotent

Reply:
@wedamija It doesn't need to be - this database is currently only being populated with any new/updated pinned searches, which have position = 0, so creating a new object with position = 1 should not create any conflicts. That being said, changing it to `update_or_create` wouldn't cause any harm either, and if it also has the added benefit for being idempotent in the case of a re-run, then I'm happy to change it just for that benefit. What do you think 


Also just for context since this was literally decided on this morning - The idea behind this addition is that we don't want a user's old pinned search to be the _only_ view they see when the new experience is released, so we also want to persist the current Prioritized view as a second view.